yt_idefix/data_structures.py

```diff
@@ -724,3 +797,241 @@ def _validate_units_override_keys(cls, units_override):
+    _default_definitions_header = "definitions.h"
+    _default_inifile = "pluto.ini"
+    _field_offset_index = None
+
+    def _parse_parameter_file(self):
+        grid_file = os.path.join(os.path.dirname(self.parameter_filename), "grid.out")
+        self.parameter_filename[:-2] + "xmf"
+        out_file = os.path.join(
+            os.path.dirname(self.parameter_filename),
+            self.parameter_filename[-6:] + ".out",
+        )
+        self._default_inifile = os.path.join(
+            os.path.dirname(self.parameter_filename), self._default_inifile
+        )
+        self._default_definitions_header = os.path.join(
+            os.path.dirname(self.parameter_filename), self._default_definitions_header
+        )
+        with open(grid_file) as gridtxt:
+            txt = gridtxt.readlines()
+            self.domain_left_edge = np.array(
+                [
+                    float(
+                        txt[i]
+                        .replace(",", "")
+                        .replace("[", "")
+                        .replace("]", "")
+                        .split()[2]
+                    )
+                    for i in [6, 7, 8]
+                ]
+            )
+            self.domain_right_edge = np.array(
+                [
+                    float(
+                        txt[i]
+                        .replace(",", "")
+                        .replace("[", "")
+                        .replace("]", "")
+                        .split()[3]
+                    )
+                    for i in [6, 7, 8]
+                ]
+            )
+            self.dimensionality = int(txt[4][-2])
+            self.domain_dimensions = np.array(
+                [
+                    int(
+                        txt[i]
+                        .replace(",", "")
+                        .replace("[", "")
+                        .replace("]", "")
+                        .split()[4]
+                    )
+                    for i in [6, 7, 8]
+                ][: self.dimensionality]
+            )
+
+            def parse_geometry(geom: str):
+                import yt
+
+                if yt.version_info[:2] > (4, 1):
+                    try:
+                        from yt.geometry.api import (
+                            Geometry,  # type: ignore [attr-defined]
+                        )
+
+                        return Geometry(geom)
+                    except ImportError:
+                        pass
+
+                return geom
+
+            geom_str = (txt[5].split()[-1]).lower()
+            self.geometry = parse_geometry(geom_str)
+
+        with open(out_file) as outttxt:
+            txt = outttxt.readlines()
+            entry = int(
+                os.path.basename(self.parameter_filename)
+                .replace(".flt.h5", "")
+                .replace("data.", "")
+                .replace(".dbl.h5", "")
+            )
+            self.current_time = float(txt[entry].split()[1])
```

Comment:
again, can you explain how many fields are expected from that `split` class and what they are supposed to be ?

Reply:
The filenames are data.%04d.dbl.h5 or data.%04d.flt.h5 
However, I didn't want the code to depend on %04d but to have something more general.

Reply:
this warrants a comment then

Reply:
Added!